server/data/ideas.js

```diff
@@ -1,111 +1,286 @@
+// ======================================================================
 const { ObjectId } = require('mongodb');
 const { ideas } = require('../config/mongoCollections');
 const { isValidObjectId, isValidStr } = require('../utils');
-const { isValidLookingFor, isValidStatus } = require('../utils/ideas');
 const { isValidProjectName } = require('../utils/projects');
 const { isValidUsername } = require('../utils/users');
 const { isValidTechnologies } = require('../utils/projects');
-const {
-	internalServerErr,
-	badRequestErr,
-	notFoundErr,
-} = require('../utils/index');
+const { internalServerErr, badRequestErr, notFoundErr } = require('../utils/index');
 
+const helper = require('../utils/ideas');
+
+// getIdeaById
+// ======================================================================
 const getIdeaById = async (idParam) => {
 	const id = isValidObjectId(idParam);
+
 	const ideasCollection = await ideas();
+
 	const idea = await ideasCollection.findOne({ _id: ObjectId(id) });
-	if (!idea) throw notFoundErr('No project found for the provided id');
+	if (!idea) throw notFoundErr('No Idea found for the provided id');
+
 	return idea;
+
 };
 
+
+// createIdea
+// ======================================================================
 const createIdea = async (ideasObjectParam, user) => {
 	const userInfo = user;
+
 	userInfo._id = ObjectId(isValidObjectId(userInfo._id));
 	userInfo.username = isValidUsername(userInfo.username);
+
 	const ideasCollection = await ideas();
-	let { name, description, technologies, lookingFor, status } =
-		ideasObjectParam;
+
+	let { name, description, technologies, lookingFor, status } = ideasObjectParam;
+
 	name = isValidProjectName(name);
 	description = ideasObjectParam.description
 		? isValidStr(ideasObjectParam.description, 'idea description')
 		: null;
 	technologies = isValidTechnologies(technologies);
-	lookingFor = isValidLookingFor(lookingFor);
-	status = isValidStatus(status);
-	const date = new Date();
+	lookingFor = helper.isValidLookingFor(lookingFor);
+	status = helper.isValidStatus(status);
+
 	const createIdeaObject = {
 		name,
 		description,
 		status,
 		lookingFor,
-		createdAt: date,
+		createdAt: new Date(),
 		technologies,
 		owner: userInfo,
 		comments: [],
 		likes: [],
 	};
-	const createIdeaAcknowledgement = await ideasCollection.insertOne(
-		createIdeaObject
-	);
-	if (
-		!createIdeaAcknowledgement?.acknowledged ||
-		!createIdeaAcknowledgement?.insertedId
-	)
+
+	const createIdeaAck = await ideasCollection.insertOne(createIdeaObject);
+
+	if (!createIdeaAck?.acknowledged || !createIdeaAck?.insertedId) {
 		throw internalServerErr('Could not create idea. Please try again');
+	}
+
 	const createdIdea = await getIdeaById(
-		createIdeaAcknowledgement.insertedId.toString()
+		createIdeaAck.insertedId.toString()
 	);
+
 	return createdIdea;
```

Comment:
Bro.. what is the point of making createIdeaAcknowledgment into createIdeaAck ??